cmd/goplugin-heroku/resource/build.go

```diff
@@ -253,3 +264,79 @@ func uploadSource(filePath, httpMethod, httpUrl string) error {
 
 	return nil
 }
+
+// This function loops through any Buildpacks defined in the Build and compares
+// them with the Buildpacks that exist in Heroku already. The status slice tracks
+// the list of results, which are then compared against each other for discrepencies.
+// If any elements of the slice are different from any other, the function returns false
+// telling the Update method that the Build needs to be updated.
+func buildpackCheck(desiredState Build, currentState heroku.Build) bool {
+	var status []bool
+	var result bool
+
+	// Heroku's Build API always returns the Buildpacks in the same order
+	// they were defined. That is used to advantage here if a user has defined
+	// one or more buildpacks in their workflow.
+	for i, packs := range desiredState.Buildpacks {
+		if strings.Contains(currentState.Buildpacks[i].URL, *packs.Name) {
+			status = append(status, true)
+		} else if currentState.Buildpacks[i].URL == *packs.URL {
+			status = append(status, true)
+		} else {
+			status = append(status, false)
+		}
+	}
+
+	if len(status) < 2 {
+		return status[0]
+	} else {
+		for p := 0; p < len(status); p++ {
```

Comment:
This seems odd to me. If `status[0]` is `false`, and if there is only one status, then this function returns `false` (line 291). However, if `len(status) == 2` and both `status[0]` and `status[1]` are `false`, then this function returns `true`. Why is that?